### ISSUE TITLE
Extract Job.FileStore into separate module

### ISSUE THREAD

**Issue by arkal** [BD2KGenomics/toil#1028]
FileStore was your average guy, living in a luxurious space with a Job(.py), and a lot of responsibility. Everything changed when the new kid on the block (CachedFileStore) decided to cause trouble in his neighborhood. The new kid was smart... he observed FS and imbibed many of his properties. What the young CFS didn't realize is that these actions would bind the two forever.... or so it seemed.

Now, with the neighborhood getting too crowded, the duo decided to move to the countryside to their new estate at fileStore.py.

To be continued

TL;DR - refactoring FileStore and CachedFileStore into fileStore.py because Job is getting too large.

**Comment by hannes-ucsc** [comment]
> countryside to their new estate at fileStore.py

… where they will grow tiny organic potatoes and eat them with the skin on.

**Comment by arkal** [comment]
haha :D
